### DISCUSSION TITLE
How to create a new stateful material property at restart, which is needed by material object at restart

### DISCUSSION THREAD

**Discussion by michaelsal75** [idaholab/moose#26192] [Q&A General]
Hello,

we are trying to make a restart in which the previous simulation includes a crystal plasticity model without damage
and the current simulation includes a damage model, thus requiring an additional stateful material property, called "hist",
representing the history of the elastic energy.

In the previous simulation we use this object:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/ComputeDislocationCrystalPlasticityStress.C

In the restart simulation we use a similar one with damage:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/ComputeCrystalPlasticityStressDamage.C

In the constructor of the material class ComputeCrystalPlasticityStressDamage in the current simulation,
there is a declaration:

_H(declareProperty<Real>("hist"))

which is not present in the material object of the previous simulation.
I expected this to create a new stateful material property (initialized at 0) and to be able to restart,
the same that I would do with an auxvariable, however we get the error:

"There is a mismatch in the stateful material properties stored during checkpoint"

saying that "hist" is not available in the "Stored stateful properties" of the previous output file.

Is there a way to solve this problem?
Can we add a new stateful material property in a restart input file?

We tried to define a new material with the same name in the restart input file using GenericConstantMaterial 
but of course it complains that "hist" is declared by multiple materials.

Thank you very much for your help.

**Comment by lindsayad** [comment]
@loganharbour this error message is yours and I know you touched this stuff in your recent refactors of restart. Any suggestions on this?

**Comment by loganharbour** [comment]
Was this capability supported previously? I don't think that it was (see the comment below) https://github.com/idaholab/moose/blob/5ce9f8980a835d962e968f108f9a3299c97c1c3d/framework/src/problems/FEProblemBase.C#L1152

I'm actively working on this in https://github.com/idaholab/moose/issues/25840, but the priority isn't the highest at the moment.

**Comment by ngrilli** [comment]
@loganharbour  thank you very much for working on this feature.

I believe for the moment it will work if we manually add the material properties in the previous simulation
using GenericConstantMaterial, i.e. a material property with the same name as the one that will be used
in the new material object in the restart simulation, and then we restart with the new material object.
